src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1255,7 +1255,10 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
 
                 // Clear callback function in WinHTTP to prevent
                 // further native callbacks as we clean up the objects.
-                SetStatusCallback(requestHandle, null);
```

Comment:
@davidsh Nit: This is a huge function and this error seems to be caused by lack of nearby context. Please consider refactoring into smaller functions to avoid these kind of bugs in the future.